### PR TITLE
Move Any wrappers and Archive to stable packages

### PR DESCRIPTION
**Pull request by cocreature**
There are no actual API or functionality changes in this PR but the
logic for locating the stable packages has slightly changed since the
Any wrappers package only makes sense for LF 1.7. To address this, we
simply filter out stable packages for newer LF versions since it
doesn’t make sense to depend on those anyway.

CHANGELOG_BEGIN

- [DAML Compiler] Move ``Archive`` type to a separate DALF.

CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags, if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs [hunk 1]
```diff
@@ -485,23 +486,30 @@ generatePackageMapRule opts =
...
+            let unitId = stringToUnitId $ if packagePath == "daml-stdlib"
+                    then damlStdlib -- We patch this to add the version number
+                    else packagePath
             let moduleName = LF.ModuleName (NonEmpty.toList $ T.splitOn "-" $ T.pack $ dropExtension $ takeFileName dalf)
             dalfPkgOrErr <- readDalfPackage dalf
             pure (fmap ((unitId, moduleName),) dalfPkgOrErr)
-    pure (diags, Map.fromList pkgs)
+    -- We filter out stable packages for newer LF versions, e.g., the stable packages for wrappers around Any.
+    -- It might seem tempting to make stable packages per LF version but this makes no sense at all.
+    -- Packages should remain stable as we move to newer LF versions.
```
Comment (hurryabit) [inline compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs]: Maybe it's good to clarify that changing the LF version would change the hash. For me, that's the main reason why having stable packages per LF version doesn't make sense.

File: compiler/damlc/stable-packages/BUILD.bazel [hunk 2]
```diff
@@ -30,12 +30,16 @@ genrule(
...
         "daml-prim/DA-Types.dalf",
+        "daml-stdlib/DA-Internal-Template.dalf",
+        "daml-stdlib/DA-Internal-Any.dalf",
     ],
     cmd = """
       $(location :generate-stable-package) --module GHC.Types -o $(location daml-prim/GHC-Types.dalf)
       $(location :generate-stable-package) --module GHC.Prim -o $(location daml-prim/GHC-Prim.dalf)
       $(location :generate-stable-package) --module GHC.Tuple -o $(location daml-prim/GHC-Tuple.dalf)
       $(location :generate-stable-package) --module DA.Types -o $(location daml-prim/DA-Types.dalf)
+      $(location :generate-stable-package) --module DA.Internal.Any -o $(location daml-stdlib/DA-Internal-Any.dalf)
```
Comment (hurryabit) [inline compiler/damlc/stable-packages/BUILD.bazel]: Do theis need to be stable given that the `Any*` types are not serializable?

File: compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs (hunk 1 above)
Comment (cocreature) [reply compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs]: Good idea, I’ve added a comment.

File: compiler/damlc/stable-packages/BUILD.bazel (hunk 2 above)
Comment (cocreature) [reply compiler/damlc/stable-packages/BUILD.bazel]: Yes, since they leak into typeclass signatures.

Comment (hurryabit) [reply compiler/damlc/stable-packages/BUILD.bazel]: Very good point. I'm glad you spotted that. Kudos!

**Pull request merged**